### PR TITLE
Dev sections order

### PR DESCRIPTION
**Pull request by alex-tinarsky**
Added SectionsOrderRule

### CODE REVIEW

File: src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt [hunk 1]
```diff
@@ -190,6 +191,49 @@ val RULE_SYMBOLS_IN_SECTION_NAMES = TableOfContentRuleBuilder()
 	}.called("""Символы ":", ".", "," в названии секции""")
 	.getRule()
 
+val sectionsThatMayPrecedeThis = mapOf<String, HashSet<String>>(
+	"Введение" to hashSetOf(""),
```
Comment (Darderion) [inline src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt:195]: Could be replaced with INTRODUCTION as it is used in multiple places

File: src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt [hunk 2]
```diff
@@ -190,6 +191,49 @@ val RULE_SYMBOLS_IN_SECTION_NAMES = TableOfContentRuleBuilder()
...
+	"Постановка задачи" to hashSetOf("Введение"),
```
Comment (Darderion) [inline src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt:196]: https://github.com/Darderion/map/pull/9#discussion_r824535710

File: src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt [hunk 3]
```diff
@@ -190,6 +191,49 @@ val RULE_SYMBOLS_IN_SECTION_NAMES = TableOfContentRuleBuilder()
...
+)
+
+val RULE_SECTIONS_ORDER = TableOfContentRuleBuilder()
+	.disallow { listOfLines ->
+		var nameOfPreviousSection = ""
+		listOfLines
+			.filterNot { line ->
+				val words = line.text
+					.filter { it.text.trim().isNotEmpty() }
+					.filterNot { it.text.contains("[0-9]+\\.".toRegex()) }
```
Comment (Darderion) [inline src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt:210]: https://github.com/Darderion/map/pull/18#discussion_r842637313

**Review by Darderion — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt (hunk 3 above)
Comment (alex-tinarsky) [reply src/main/kotlin/com/github/darderion/mundaneassignmentpolice/rules/Rules.kt]: Done

**Pull request merged**